workitem/enum_type_blackbox_test.go

```diff
@@ -321,3 +321,86 @@ func TestEnumType_EqualEnclosing(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumType_ConvertFromModel(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	type testCase struct {
+		subTestName    string
+		input          interface{} // contains valid and invalid values
+		expectedOutput interface{}
+		wantErr        bool
+	}
+	tests := []struct {
+		name string
+		enum w.EnumType
+		data []testCase
+	}{
+		{
+			"kind string",
```

Comment:
We would need similar test data for `ConvertToModel` method but I am not sure how we could separate data from the test.

Reply:
That will be part of first cleaning up the typesystem. The test data needs to know no only input and output but also how a value is stored inside:

JSON -> Go -> DB -> Go -JSON